fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -185,11 +198,9 @@ class RYWImpl {
 		return key;
 	}
 
-	ACTOR template <bool backwards>
-	static Future<RangeResult> readThrough(ReadYourWritesTransaction* ryw,
-	                                       GetRangeReq<backwards> read,
```

Comment:
How related is this change from a templated function on `GetRangeReq<backwards>` to a non-templated version? I'm not entirely sure why this was templated before, instead of doing what you're doing here, but there is potentially something to be gained from optimizations that can be done on templated code (for example, at least one `if` clause can be simplified to `if (false)` and thus compiled away).

Reply:
this is a blocker of my change , see https://github.com/apple/foundationdb/pull/9257/files#diff-a5fc2f5c1e21e55b139e63e9e4d17951214b29f07966e92bc3c2c6770dd0aa05R1206-R1211(it seems we cannot open it directly), line 1206-1211 in `fdbclient/ReadYourWrites.actor.cpp`.

by removing it, caller of `readThrough` can have template, although we still have 2 `readThrough`, but we have only 1 caller.  Without this change, the parameter of caller is `GetMappedRangeReq<backwards> req`, and unfortunately I cannot use GetMappedRangeReqV2 as a type here.